### ISSUE TITLE
Passing Redis client to config causes "Maximum call stack size exceeded" with deep-extend

### ISSUE THREAD

**Issue by SethTippetts** [stormpath/stormpath-sdk-node#198]
Seems to happen on line 43 of `lib/Config.js`:

`deepExtend(this, options||{});`

Here is my code:
```js
import express from 'express';
import path from 'path';
import config from 'config';
import redis from 'redis';
import { init as stormpath } from 'express-stormpath';

let app = express();

let client = redis.createClient(config.get('redis.port'), config.get('redis.host'), {
  auth_pass: config.get('redis.password')
});

app.use(stormpath(app, {
  api: true,
  website: true,
  expandGroups: true,
  registrationView: path.join(__root, 'views/register.jade'),
  loginView: path.join(__root, 'views/login.jade'),
  cacheOptions: {
    store: 'redis',
    client,
    ttl: 300,
    tti: 300
  },
  getOauthTokenUrl: '/oauth/token',
  oauthTTL: 5 * 60 // 5 minutes
}));
```

If it's an issue with my implementation and not the code, updating the docs would be helpful.

I'm using Heroku Redis which has a connection limit of 25. Using the default redis config for stormpath seemed to break that limit every time I ran the app. I ran into this issue while trying to use a custom client. Thanks in advance!

**Comment by robertjd** [comment]
@typerandom can you investigate this?  Thanks!

**Comment by typerandom** [comment]
@robertjd Yes, I'm looking into this now.

**Comment by typerandom** [comment]
The issue here is that the `deep-extend` module that we're using is trying to copy the configuration object passed to the client and because it references the Redis client that contains references of itself, a circular dependency occurs. And when this is traversed it eventually crashes when it reach the max stack size. So I'm right now looking into other extend libraries, or how we can patch the `deep-extend` module so that we can have values in the configuration that it avoids copying. We could also do detection of circular dependencies, but I'd rather avoid that if possible.

**Comment by typerandom** [comment]
There's now a PR in our `stormpath-config` module repo that should fix this. So once it's merged and we have a new `stormpath-config` release, then we'll just update to that version, and that should fix this error. The solution was pretty simple. I just switched from using the `deep-extend` to the `cloneextend` module (which does the same as `deep-extend` but supports circular dependencies).

**Comment by typerandom** [comment]
I have now updated the package to use the latest `stormpath-config` version (v0.0.7). Closing issue.

**Comment by typerandom** [comment]
@SethTippetts I just released a new SDK version (v0.13.2). This should resolve this error. Please try it out and let me know how it works for you.
